src/taipy/gui_core/GuiCoreLib.py

```diff
@@ -253,7 +257,42 @@ def crud_scenario(self, state: State, id: str, action: str, payload: t.Dict[str,
                 state.assign(_GuiCoreContext._SCENARIO_SELECTOR_ERROR_VAR, f"Invalid date ({date_str}).{e}")
                 return
             try:
-                scenario = create_scenario(scenario_config, date, name)
+                do_create = True
+                on_creation = args[3] if len(args) > 3 and isinstance(args[3], str) else None
+                on_creation_function = gui._get_user_function(on_creation) if on_creation else None
+                if callable(on_creation_function):
+                    try:
+                        res = gui._call_function_with_state(
+                            on_creation_function,
+                            [
+                                id,
+                                on_creation,
+                                {
+                                    "config": scenario_config,
+                                    "date": date,
+                                    "name": name,
+                                    "properties": data.get("properties"),
+                                    "before": True,
```

Comment:
Do you still need this, with all the paths from on_creation?

Reply:
I don't understand your comment

Reply:
do you mean that if there's no scenario it is before and if there's a scenario it is after ?
I think adding before(bool) makes things easier in the callback:
`if payload.get("before", False):`

Reply:
Sorry, I was not clear. 

What I think is that on_creation() would be invoked right after the Create button is pressed.
That would send all that is needed to create the scenario, as I see in the code.
Now the user can add, check, modify these parameters, and decide what to do next:
- No changes whatsoever: return None. The back end (of GUI) can now create the scenario as of today, and nothing else.
- There are things to set before the creation or to change afterward: the user creates the scenario himself, invoking the Core API.
   With the scenario in hand, it can perform whatever need to be done, then return it.
    The back end (of GUI) receives the scenario, and processes as if it had created it itself.
 - No scenario should be created at all. May be parameters are wrong, maybe there are already too many scenarios... nobody knows except the programmer.
    The programmer then fills in some explanatory text, and return it.
    The back end (of GUI) receives the string, and probably triggers a notification with the text (if not empty) and nothing else.
    
 Basically, as long as there is no changes to perform things work as "before" 2.4.
 If anything has to be setup before the scenario creation, or after, then the user is on his own.  He will certainly have to call `create_scenario(scenario_config, creation_date, scenario_label)` with the parameters received by the callback, and set the properties manually (they are in the callback parameters as well).
 This can be shown in a short example in the doc. We can even think of adding a scenario creation API with the CoreGui lib to simplify that creation if needed. But Core should have that of course.

Reply:
OK I get it